src/extra/projjson.jl

```diff
@@ -0,0 +1,360 @@
+# Prefer a json structure that is more inline with GDAL's or our WKT input
+const GDALcompat = true
+
+get_main_key(d) = nothing
+function get_main_key(d::Dict)
+  @assert d |> keys |> length == 1
+  return d |> keys |> first
+end
+
+# From a Vector{Dict}, get Dict items that all has a particular key  
+function get_items_with_key(key::Symbol, list::Vector)
+  filter(x->get_main_key(x)==key, list)
+end
+
+function get_item_with_key(key::Symbol, list::Vector)
+  found = filter(x->get_main_key(x)==key, list)
+  return isempty(found) ? nothing : found
+end
+
+function get_item_with_keys(keys::Vector{Symbol}, list::Vector)
+  found = []
+  for key in keys
+    f = get_item_with_key(key, list)
+    !isnothing(f) && append!(found, f)
+  end
+
+  if length(found) == 1
+    return found[1]
+  elseif length(found) == 0
+    return nothing
+  elseif length(found) > 1
+    error("Multiple items found for keys: $keys")
+  end
+end
+
+# --------------------------------------
+# Convert From WKT dict to projjson dict
+# --------------------------------------
+
+function wkt2json(wkt::Dict)
+  type = get_main_key(wkt)
+  if type == :GEOGCRS
+    return wkt2json_geog(wkt)
+  elseif type == :GEODCRS
+    return wkt2json_geog(wkt)
+  elseif type == :PROJCRS
+    return wkt2json_proj(wkt)
+  else
+    error("Unimplemented CRS type: $type")
+  end
+end
+
+# process GEOGCRS or GEODCRS nodes, either if they're at top-level or under PROJCRS
+function wkt2json_geog(wkt::Dict)
+  @assert wkt |> keys |> collect |> first in [:GEOGCRS, :GEODCRS]
+  jsondict = Dict{String, Any}()
+  geo_subtype = get_main_key(wkt) 
```

Comment:
Please avoid "_" in variable and function names.

Reply:
👍🏼 many of them is easy.

but what do you recommendnd for the `wkt2json_*` functions?

Reply:
I will read the code carefully and get back with possible names 👍🏽